src/main/java/org/radarcns/phone/PhoneSensorManager.java

```diff
@@ -304,13 +309,12 @@ private void processBatteryStatus(Intent intent) {
         getState().setBatteryLevel(batteryPct);
 
         double time = System.currentTimeMillis() / 1000d;
-        trySend(batteryTopic, 0L, new PhoneBatteryLevel(
-                time, time, batteryPct, isPlugged, batteryStatus));
+        send(batteryTopic, new PhoneBatteryLevel(time, time, batteryPct, isPlugged, batteryStatus));
```

Comment:
Maybe the problem is not the broadcast listener, but this single line. When we wrote this, we thought that battery level is not going to be very important. The trySend method tries to send data, but if there is not internet or the battery level is low, it will not be sent. The data is also not cached, so if a request fails, the data is also not sent. Can we try just changing this line, and see if it already improves things?